src/crimson/osd/ops_executer.cc

```diff
@@ -970,20 +970,28 @@ std::unique_ptr<OpsExecuter::CloningContext> OpsExecuter::execute_clone(
   osd_op_params->at_version.version++;
   encode(cloned_snaps, cloning_ctx->log_entry.snaps);
 
-  // update most recent clone_overlap and usage stats
-  assert(cloning_ctx->new_snapset.clones.size() > 0);
-  // In classic, we check for evicted clones before
-  // adjusting the clone_overlap.
-  // This check is redundant here since `clone_obc`
-  // was just created (See prepare_clone()).
-  interval_set<uint64_t> &newest_overlap =
-    cloning_ctx->new_snapset.clone_overlap.rbegin()->second;
-  osd_op_params->modified_ranges.intersection_of(newest_overlap);
-  delta_stats.num_bytes += osd_op_params->modified_ranges.size();
-  newest_overlap.subtract(osd_op_params->modified_ranges);
   return cloning_ctx;
 }
 
+void OpsExecuter::update_clone_overlap() {
+  if (op_info.may_write() && obc->obs.exists && !snapc.snaps.empty()) {
+    interval_set<uint64_t> *newest_overlap;
+    if (cloning_ctx) {
+      newest_overlap =
+      &cloning_ctx->new_snapset.clone_overlap.rbegin()->second;
+    } else if (!obc->ssc->snapset.clones.empty()) {
+      newest_overlap =
+      &obc->ssc->snapset.clone_overlap.rbegin()->second;
+    } else {
+      return;
+    }
+    if (osd_op_params) {
```

Comment:
Moreover, if we have `cloning_ctx` value, then `want_mutate` should also be true. Meaning we must also have have `osd_op_params`. We should be able to assert `osd_op_params` here.

Reply:
`cloning_ctx` is non-null requires `snapc.snaps[0] > initial_obc.ssc->snapset.seq` is true, but when write op occur and `snapc.snaps[0] > initial_obc.ssc->snapset.seq` is false like [source](https://github.com/ceph/ceph/blob/main/src/test/librados/snapshots_cxx.cc#L283), `cloning_ctx` will be null, but we also need update `clone_overlap`

Reply:
yes,  We should be able to assert osd_op_params here, because write op always initialize `osd_op_params`